_posts/2018-05-23-getting-foreman-search-results-into-your-puppet-manifest.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: update - Getting foreman search results into your Puppet manifest
+date: 2018-05-23 11:19:42
+author: Fabien Combernous
+tags:
+- foreman
+- puppet
+- puppetdb
+- monitoring
+- backup
+- identity management
+---
+
+How to use the foreman search language inside puppet manifests and knowing informations about others nodes.
+This page is an update of the [previous post written by Ohad Levy](https://theforeman.org/2012/01/getting-foreman-search-results-into.html)
+
+<!--more-->
+
+## Why ?
+
+When you start puppet, in general you start simply. And so innocently, each node is more or less alone in the world. And puppet, in this way, can already help you so much.
+
+But fastly, you understand that in some situations, it is interesting to know informations from others nodes. And this happend tipically about monitoring, backups or identity management. For instance, you add a new node into your puppet infrastructure. And your monitoring server takes into account this new server by adding the configuration relating to its monitoring. 
+
+In a standard way, in the puppet galaxy it exists the option `storeconfigs` of pupper server and linked [puppetdb server](https://puppet.com/docs/puppetdb/latest). A puppetdb server records every node's catalogs, facts and reports. And the immediate usage is exported resources (a double “at” sign `@@` in manifests). 
```

Comment:
typo, pupper -> puppet. I would also recommend using Puppet, Puppet Server and PuppetDB when referring to the product, `puppet` when referring to the call.

Reply:
Same goes for Foreman, capitalized